### DISCUSSION TITLE
Split dividers are hard to see

### DISCUSSION THREAD

**Discussion by david-crespo** [ghostty-org/ghostty#3301] [Feature Requests, Ideas] [answered]
The [logic](https://github.com/ghostty-org/ghostty/blob/a8e5eef11cc67f87f445626f9ca2993373774bf8/macos/Sources/Ghostty/Ghostty.Config.swift#L364-L371) for split divider color (darken bg by 0.08 if dark, otherwise by 0.4) results in barely visible lines, on dark themes especially.

Once I found the code and therefore the name "split divider", I was able to find some [discussion](https://discord.com/channels/1005603569187160125/1255201317505335346/1255201317505335346) about this on Discord. I figure it's worth moving the discussion here. It sounded like the most likely solution is to make the color configurable:

> mitchellh: We could easily add a config to modify this color manually

as opposed to, for example, making the logic overly clever.

### Example (ayu theme)

There is a horizontal and a vertical divider here, both nearly invisible. With an even darker background, it's totally invisible, because of course darkening black gives you black.

<img width="1014" alt="image" src="https://github.com/user-attachments/assets/896bbd1e-e039-4fb3-befe-02ca99179187" />

**Comment by khaledh** [comment]
I'd love to also have the divider width configurable. This is what it looks like in iTerm:

<img src="https://github.com/user-attachments/assets/d2bbb35e-5ada-447f-b875-270d4795d500" width="800">

**Reply by azasypkin** [reply]
Yeah, that would be ideal. With so many people having various color vision deficiencies, it would be really hard to come up with the auto-detectable color for the divider that works for everyone (I couldn’t achieve a satisfactory result for myself with just a color). Offering both configurable width and color should address this for almost everyone.

**Comment by ghost** [comment]
A contrasting visible line (color configurable) is enough. It would help to avoid eye strain.
I find a too-wide divider very ugly, but others might think differently.

**Comment by goproslowyo** [comment]
Everyone who's chimed in here so far seems to be on macOS but for the Linux folks that want to adjust this for now use custom css for gtk.

Extreme ugly example to demonstrate:
![image](https://github.com/user-attachments/assets/3e4fc730-dcc3-463f-a7c1-61e15934c052)

 https://discord.com/channels/1005603569187160125/1322959239693733972/1323020932549312592

**Comment by jjt** [comment]
> mitchellh: We could easily add a config to modify this color manually

going with the Ghostty philsophy of zero-config, might be best to detect the bg color and darken or lighten depending

or could do two tone (if feasible in UI code) of lightened bg line above darkened bg line, like this mockup, might work on any theme?

edit: for horizontal splits, would have the light part on the left, as per humans' preference for [top-left lighting](https://en.wikipedia.org/wiki/Top-left_lighting), assuming we want the split to appear like a convex "bump" outward - reverse the order for a concave "divot"

![Group 1](https://github.com/user-attachments/assets/b40fc413-5de5-47e2-8382-7226630def3a)

**Reply by david-crespo** [reply]
Surely worth considering, though I suspect it will be hard to come up with logic so good that it doesn't also have to be made configurable.

**Comment by lcrespom** [comment]
I have a completely black background, so in my case the split divider is completely invisible.
My workaround is to fill the unfocused split with some color:
```properties
unfocused-split-fill = #500028
```

<img width="1206" alt="Screenshot 2025-01-01 at 18 36 11" src="https://github.com/user-attachments/assets/fd04f3c7-9943-4072-9162-5d3d89c45dec" />

**Reply by creechy** [reply]
I never really liked how iTerm2 changed the background like this. But not having it (by default) with ghostty, I realize I miss it. Thanks for the tip.

**Comment by mitchellh** [comment]
Approved manual config #4326.

I'd love to also make the logic for the automatic version better so most people _do not_ have to touch this. I'm just accepting #4326 in the short term so people have an escape hatch. To that end, are there suggestions on how to improve the current formula?

**Reply by jjt** [reply]
i feel like i might be missing something, but naively could make an `OSColor.lighten` method a la [`OSColor.darken`](https://github.com/ghostty-org/ghostty/blob/94599102e9fb8247af08cbbbcb7ee25e3d31e1bd/macos/Sources/Helpers/OSColor%2BExtension.swift#L82)?

```diff
var splitDividerColor: Color {
    let backgroundColor = OSColor(backgroundColor)
    let isLightBackground = backgroundColor.isLightColor
-   let newColor = isLightBackground ? backgroundColor.darken(by: 0.08) : backgroundColor.darken(by: 0.4)
+   let newColor = isLightBackground ? backgroundColor.darken(by: 0.08) : backgroundColor.lighten(by: 0.08)
    return Color(newColor)
}
```

**Reply by J-Siu** [reply]
@mitchellh IMHO, I understand the concept of zero config and everything should work out of the box, which Ghostty is already doing well. However, I think you are a bit over-thinking here.

If users are changing background color, transparency, etc., they are touching the config, and options like `split-divider-color` are a requirement, not just a temporary solution.

Especially with transparency, desktop wallpaper/color, overlapping with other applications, etc. The possibility is endless. Just let the users play around with the option. And even you add some magical desktop detection into Ghostty, someone will still ask for the option to set their own color🤣.

**Reply by david-crespo** [reply]
Making it configurable isn’t temporary — I think “short term” here means “quick”. He just wants to make it less likely that someone needs to configure it.

**Reply by Tener** [reply]
> To that end, are there suggestions on how to improve the current formula?

How about defaulting to foreground text color (possibly with reduced saturation)? This is a setting that people pay a lot of attention to in their themes and is sure to be highly visible.

**Reply by jjt** [reply]
foreground color should be for foreground text, not a UI affordance - basing it on some part of the background/chrome/etc seems more appropriate here

configurable setting is definitely needed though, lot of valid and differing preferences 🙂

**Reply by J-Siu** [reply]
Maybe we can just make it simple as what MacOS is doing with the window border:

![Screenshot 2025-01-11 at 19 18 56](https://github.com/user-attachments/assets/a48bf2a3-50a9-479d-885f-9b4314351281)

If option is not set, and dark theme/background is used, divider is white, else it is black.

Then user can customize it with an option.

**Reply by jcollie** [reply, marked as answer]
There will be an option to configure the divider color in the next release.

**Comment by doums** [comment]
Hey, sorry to re-ping that discussion but working with high DPI monitors I find the split divider lines way too thin like almost invisible (1 pixel is really minuscule on 4K). Looking at the config reference, now there is the option to change its color but sadly nothing for the thickness.
As a suggestion, I think it would be a helpful option to add to the config.

**Comment by zmunk** [comment]
Ok, the split divider color can be changed like so:
```
split-divider-color = #F0f0f0
```

[Source](https://ghostty.org/docs/install/release-notes/1-1-0)

**Comment by davelnewton** [comment]
Not allowing width configuration is bad

**Comment by 00-kat** [comment]
Let's move discussion on making the split divider thicker to a new discussion (if one doesn't exist already for that specific purpose).

**Comment by curehabit** [comment]
You can set the color of split line by add this to ~/.config/ghostty/config
`split-divider-color = #27f549`

**Reply by davelnewton** [reply]
Yes, it says that above 🤷‍♂️
